Ensure test is suffixed with Test

greg0ire opened the pull request:
In #8591 , I noticed that some tests were not run, and found an exhaustive list of tests that need to be renamed. Some did not require much modifications to run again, but that last one does not pass in the CI, although it passes locally on my machine.

The assertion that fails is about an entity fetched through a one-to-one relationship. It is not supposed to be a proxy but on the CI, it seems to be.

The test that fails looks like this: https://github.com/doctrine/orm/blob/7de84537f61129a96658e50799131dec779c55bd/tests/Doctrine/Tests/ORM/Functional/ClassTableInheritanceTest2.php#L38-L62

The failure looks like this: 

```
1) Doctrine\Tests\ORM\Functional\ClassTableInheritanceSecondTest::testOneToOneAssocToBaseTypeBidirectional
Failed asserting that Doctrine\Tests\Proxies\__CG__\Doctrine\Tests\ORM\Functional\CTIChild Object (...) is not an instance of interface "Doctrine\ORM\Proxy\Proxy".
```

greg0ire committed:
Ensure test is suffixed with Test

stof commented:
why would we enforce that it is not a proxy ? Instantiating a proxy object or the entity class itself is internal details of the hydrator to me. It could instantiate a proxy even if it initializes it directly.

greg0ire commented:
Ok, works for me, I'll remove that assertion then 👍

greg0ire committed:
Drop assertion about not being an instance of proxy

We do not want to enforce it as it is an internal details that seems to
vary from environment to environment.
